src/Lib.P2P/ConnectionManager.cs

```diff
@@ -0,0 +1,252 @@
+
+        /// <summary>
+        ///   Raised when a peer's connection is closed.
+        /// </summary>
+        public event EventHandler<MultiHash> PeerDisconnected;
+
+        /// <summary>
+        ///   Gets the current active connections.
+        /// </summary>
+        public IEnumerable<PeerConnection> Connections =>
+            _connections.Values
+               .SelectMany(c => c)
+               .Where(c => c.IsActive);
+
+        /// <summary>
+        ///   Determines if a connection exists to the specified peer.
+        /// </summary>
+        /// <param name="peer">
+        ///   Another peer.
+        /// </param>
+        /// <returns>
+        ///   <b>true</b> if there is a connection to the <paramref name="peer"/> and
+        ///   the connection is active; otherwise <b>false</b>.
+        /// </returns>
+        public bool IsConnected(Peer peer) { return TryGet(peer, out _); }
+
+        /// <summary>
+        ///    Gets the connection to the peer.
+        /// </summary>
+        /// <param name="peer">
+        ///   A peer.
+        /// </param>
+        /// <param name="connection">
+        ///   The connection to the peer.
+        /// </param>
+        /// <returns>
+        ///   <b>true</b> if a connection exists; otherwise <b>false</b>.
+        /// </returns>
+        /// <remarks>
+        ///   If the connection's underlaying <see cref="PeerConnection.Stream"/>
+        ///   is closed, then the connection is removed.
+        /// </remarks>
+        internal bool TryGet(Peer peer, out PeerConnection connection)
+        {
+            connection = null;
+            if (!_connections.TryGetValue(Key(peer), out var conns))
+            {
+                return false;
+            }
+
+            connection = conns
+               .FirstOrDefault(c => c.IsActive);
+
+            return connection != null;
+        }
+
+        /// <summary>
+        ///   Adds a new connection.
+        /// </summary>
+        /// <param name="connection">
+        ///   The <see cref="PeerConnection"/> to add.
+        /// </param>
+        /// <returns>
+        ///   The connection that should be used.
+        /// </returns>
+        /// <remarks>
+        ///   If a connection already exists to the peer, the specified
+        ///   <paramref name="connection"/> is closed and existing connection
+        ///   is returned.
+        /// </remarks>
+        public PeerConnection Add(PeerConnection connection)
+        {
+            if (connection == null)
+            {
+                throw new ArgumentNullException(nameof(connection));
+            }
+
+            if (connection.RemotePeer == null)
+            {
+                throw new ArgumentNullException(nameof(connection.RemotePeer));
+            }
+
+            if (connection.RemotePeer.Id == null)
+            {
+                throw new ArgumentNullException(nameof(connection.RemotePeer.Id));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The parameter name '' is not declared in the argument list.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)